src/topology/algebra/ordered/basic.lean

```diff
@@ -1490,14 +1490,17 @@ local notation `|` x `|` := abs x
 lemma nhds_eq_infi_abs_sub (a : α) : 𝓝 a = (⨅r>0, 𝓟 {b | |a - b| < r}) :=
 begin
   simp only [le_antisymm_iff, nhds_eq_order, le_inf_iff, le_infi_iff, le_principal_iff, mem_Ioi,
-    mem_Iio, abs_sub_lt_iff, @sub_lt_iff_lt_add _ _ _ _ a, @sub_lt _ _ a, set_of_and],
+    mem_Iio, abs_sub_lt_iff, @sub_lt_iff_lt_add' _ _ _ _ a, @sub_lt _ _ _ _ a, set_of_and],
   refine ⟨_, _, _⟩,
   { intros ε ε0,
-    exact inter_mem_inf_sets
-      (mem_infi_sets (a - ε) $ mem_infi_sets (sub_lt_self a ε0) (mem_principal_self _))
-      (mem_infi_sets (ε + a) $ mem_infi_sets (by simpa) (mem_principal_self _)) },
+     refine inter_mem_inf_sets
+      (mem_infi_sets (a - ε) $ mem_infi_sets (sub_lt_self a ε0) _)
+      (mem_infi_sets (ε + a) $ mem_infi_sets (by simpa) _),
+    exact mem_principal_sets.mp (λ h, sub_lt_iff_lt_add'.mp),
+    exact mem_principal_sets.mp (λ x, sub_lt_iff_lt_add.mpr) },
   { intros b hb,
-    exact mem_infi_sets (a - b) (mem_infi_sets (sub_pos.2 hb) (by simp [Ioi])) },
+    exact mem_infi_sets (a - b) (mem_infi_sets (sub_pos.2 hb) (λ x h,
+      (sub_sub_cancel _ _).symm.le.trans_lt (sub_lt_iff_add.mpr h.1))) },
```

Comment:
Why is `simp` no longer able to close these goals?

Reply:
I am not sure what caused `simp` to not solve these goals.  However, I think that I managed to fix the `linarith` problem!

If/once `linarith` is fixed, I will see if marking some different lemma as `simp` will solve this.  It might have to do with the fact that one of the lemmas that was marked as `simp` involved using `rw add_comm` somewhere and I made the `simp` lemma out of the one without the `add_comm` factored in.  I will investigate!